File: 자료구조/빽준_10845_큐/큐.cpp
#include <iostream>
#include <cstring>
using namespace std;

int arr[10001];

int main() {
    int n = 0;
    scanf("%d", &n);
    char s[10];
    int f = 0;
    int b = 0;

    while(n--){
        scanf("%s", &s);
        if(strcmp(s, "push") == 0){
            int num = 0;
            scanf("%d", &num);
            arr[b] = num;
            b++;
        }
        else if(strcmp(s, "pop") == 0){
            if(f == b) printf("-1\n");
            else{
                printf("%d\n", arr[f]);
                arr[f] = 0;
                f++;
            }
        }
        else if(strcmp(s, "size") == 0) printf("%d\n", b-f);
        else if(strcmp(s, "empty") == 0){
            if(f == b) printf("1\n");
            else printf("0\n");
        }
        else if(strcmp(s, "front") == 0){
            if(f == b) printf("-1\n");
            else printf("%d\n", arr[f]);
        }
        else if(strcmp(s, "back") == 0){
            if(f == b) printf("-1\n");
            else printf("%d\n", arr[b-1]);
        }
    }
}

File: 자료구조/빽준_10845_큐/큐.kt
import java.io.*
import java.util.*

fun main(args: Array<String>) {
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val s = Scanner(System.`in`)
    var n = s.nextInt()
    var arr = IntArray(10001, {0})
    var f = 0
    var b = 0
    while(n-- > 0){
        var str = s.next()
        if(str.equals("push")){
            var num = s.nextInt()
            arr[b] = num
            b++
        }
        else if(str.equals("pop")){
            if(f == b) bw.write("-1\n");
            else{
                bw.write(arr[f].toString() + "\n")
                arr[f] = 0
                f++
            }
        }
        else if(str.equals("size")) bw.write((b-f).toString() + "\n")
        else if(str.equals("empty")){
            if(f == b) bw.write("1\n")
            else bw.write("0\n")
        }
        else if(str.equals("front")){
            if(f == b) bw.write("-1\n")
            else bw.write(arr[f].toString() + "\n")
        }
        else if(str.equals("back")){
            if(f == b) bw.write("-1\n")
            else bw.write(arr[b-1].toString() + "\n")
        }
    }
    bw.flush()
    bw.close()
}